File: src/main/java/com/uwetrottmann/trakt5/enums/SortHow.java
package com.uwetrottmann.trakt5.enums;

import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

public enum SortHow implements TraktEnum {
    ASC("asc"),
    DESC("desc");

    private final String value;

    SortHow(String value) {
        this.value = value;
    }

    private static final Map<String, SortHow> STRING_MAPPING = new HashMap<>();

    static {
        for (SortHow via : SortHow.values()) {
            STRING_MAPPING.put(via.toString().toUpperCase(Locale.ROOT), via);
        }
    }

    public static SortHow fromValue(String value) {
        return STRING_MAPPING.get(value.toUpperCase(Locale.ROOT));
    }

    @Override
    public String toString() {
        return value;
    }

}

File: src/main/kotlin/com/redissi/trakt/enums/SortHow.kt
package com.redissi.trakt.enums

import java.util.*

enum class SortHow(private val value: String) : TraktEnum {
    ASC("asc"), DESC("desc");

    companion object {
        private val STRING_MAPPING: MutableMap<String, SortHow> =
            HashMap()

        fun fromValue(value: String): SortHow? {
            return STRING_MAPPING[value.toUpperCase(Locale.ROOT)]
        }

        init {
            for (via in values()) {
                STRING_MAPPING[via.toString().toUpperCase(Locale.ROOT)] = via
            }
        }
    }

    override fun toString(): String {
        return value
    }

}